pusher_tests/test_crypto.py

```diff
@@ -0,0 +1,169 @@
+        fuzzy_testcases = []
+        with open('pusher_tests/fuzzy-encryption-keys.json') as f:
+            fuzzy_testcases = json.load(f)
+
+        testcases = [
+          { "input":"aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa", "expected":True },
+          { "input":"                                ", "expected":True },
+          { "input":"private--encrypted--djsahajksha-", "expected":True },
+          { "input":"dadasda;enc   ted--djsahajkshdak", "expected":True },
+          { "input":"--====9.djsahajkshdak", "expected":False }
+        ]
+
+        # merge specific testcases with the fuzzy ones
+        testcases += fuzzy_testcases
+
+        # do the actual testing
+        for t in testcases:
+          self.assertEqual(
+              crypto.is_encryption_master_key_valid( t["input"] ),
+              t["expected"]
+          )
+    
+    def test_generate_shared_secret(self):
+        
+        # load sample channel and encryption_master_key strings and their sha256 encodings
+        sha256s_testcases = []
+        with open('pusher_tests/sha256s.json') as f:
+            sha256s_testcases = json.load(f)
+
+        testcases = [
+          {
+            "input": ["pvUPIk0YG6MnxCEMIUUVFrbDmQwbhICXUcy",
+                      "OTk5OTk5OTk5OTk5OTk5OTk5OTk5OTk5"],
+            "expected": "709ef3090a24bfa98382cb5d025ca362"
+          },
+          {
+            "input": ["Hk4F3",
+                      "OTk5OTk5OTk5OTk5OTk5OTk5OTk5OTk5"],
+            "expected": "d0c0b6770e40738337cba018b0b7443d"
+          },
+          {
+            "input": ["kvC1JMEE4glNhEXXSLLNVX218RN3sEsAPgoUwKKSiQ7ASdOdEA",
+                      "MTIzNDU2Nzg5MDEyMzQ1Njc4OTAxMjM0"],
+            "expected": "51ca1e679fabc29f94545028d853e773"
+          },
+          {
+            "input": ["GqYByIx5JE9avvMwPgyFDIuJm019XfqFwfgahZSBcnetFwlZNnE1wnS3F6E9IJf409Hm",
+                      "AQIDBAUGBwgODxAREhMUFRYXGBkaGxwd"],
+            "expected": "f9c1c56016ea9f5081358a972877c3d7"
+          },
+          {
+            "input": ["V0CJ5U1r7t3oRcpKhhlVzN19",
+                      "woDCgcKCwoPChMKGwofCiMKJworCi8KM"],
+            "expected": "620fa93937212ff93f015c6a05c58772"
+          },
+          {
+            "input": ["kfxrSj01BBd7NF1ltXFn0Yi5b15oGFgR1BBOc",
+                      "CwwgwoXCoOGagOKAgOKAgeKAguKAg+KA"],
+            "expected": "d0a6197c14e51c1e23049cb75b8d6d8f"
+          },
+          {
+            "input": ["Ed8l",
+                      "wq3YgNiB2ILYg9iE2IXYnNud3I/hoI7i"],
+            "expected": "f525f853136c6c424e14d5cf63aaeecb"
+          } 
+        ]
+
+        # merge specific testcases with externally loaded ones
+        testcases += sha256s_testcases
+        # do the actual testing
+        for t in testcases:
+          self.assertEqual(
+              crypto.generate_shared_secret( t["input"][0], t["input"][1] ),
+              t["expected"]
+          )
```

Comment:
I don't think there's any harm having all these tests, but I'm not sure it's useful having more than 1 of them. If a test breaks for one, then I struggle to imagine how it would not break for all of them (unless there was a bug in the underlying encryption library - and I don't think it's our responsibility to test that).

Reply:
Yes, it was more for the sake of fuzzy testing than anything else. We should suppose that the underlying encryption library is tested against fuzzy strings.